### PR TITLE
GET-423 Skills Builder v3

### PR DESCRIPTION
**Pull request by cwrw**
jira-ticket: https://dfedigital.atlassian.net/browse/GET-423

* Fix skills builder bugs and add new content
* Add ability to remove job profile from skills page

### CODE REVIEW

File: app/controllers/job_profiles_controller.rb [hunk 1]
```diff
@@ -3,9 +3,24 @@ def show
     @job_profile = JobProfileDecorator.new(resource)
   end
 
+  def destroy
+    user_session.remove_job_profile(resource.id)
+
+    query_params = { search: job_profile_params[:search] }
+    query_params[:job_profile_id] = job_profile_params[:job_profile_id] unless user_session.job_profile_skills?
```
Comment (smoothcontract) [inline app/controllers/job_profiles_controller.rb]: I don't understand why this has an unless clause - is that for backwards compatibility? Might be worth adding a comment to explain this, perhaps with a TODO if it's something that needs to be cleaned up when we roll out the new skills builder feature.

File: app/controllers/skills_controller.rb [hunk 2]
```diff
@@ -39,7 +39,9 @@ def job_profiles
   end
 
   def show_skills_builder_v2
-    return redirect_to task_list_path unless job_profiles.any? && user_session.skill_ids.any?
+    return redirect_to task_list_path unless
+      skills_params[:job_profile_id] ||
```
Comment (tavy88) [inline app/controllers/skills_controller.rb]: Do you want

```ruby
skills_params[:job_profile_id].present?
```

here?

File: app/models/skills_builder.rb [hunk 3]
```diff
@@ -11,15 +11,18 @@ def initialize(skills_params:, job_profile:, user_session:)
...
-    return unless skills_params.present?
+    return unless skills_params.present? && valid?
 
     user_session.set_skills_ids_for_profile(job_profile.id, formatted_skill_params)
   end
 
   def skill_ids
-    @skill_ids ||=
+    @skill_ids ||= begin
+      return [] if skills_params.present? && !valid?
```
Comment (tavy88) [inline app/models/skills_builder.rb]: I think this can live outside the `begin` block. What do you think?

Comment (cwrw) [reply app/models/skills_builder.rb]: I think it's better to be in the begin block to be memoized as we call this method quite alot

File: app/controllers/job_profiles_controller.rb [hunk 4]
```diff
@@ -3,9 +3,29 @@ def show
     @job_profile = JobProfileDecorator.new(resource)
   end
 
+  def destroy
+    user_session.remove_job_profile(resource.id)
```
Comment (tavy88) [inline app/controllers/job_profiles_controller.rb]: Nice!

File: app/models/skills_builder.rb (hunk 3 above)
Comment (tavy88) [reply app/models/skills_builder.rb]: Ok fair.

File: spec/models/user_sessions_spec.rb [hunk 5]
```diff
@@ -246,32 +280,20 @@
 
   describe '#job_profile_skills?' do
     it 'returns true if the session already contains job profile skills' do
-      session[:job_profile_skills] = { '1' => [1] }
+      session[:job_profile_ids] = [1]
```
Comment (tavy88) [inline spec/models/user_sessions_spec.rb]: Assuming session[:job_profile_ids] is an array you can do 

```ruby
session[:job_profile_ids] << 1
```

here?

Comment (cwrw) [reply spec/models/user_sessions_spec.rb]: I chose to explicitly do this as this means the job profile ids array only equals 1 as opposed to appending it to any existing array with values

**Pull request merged**